packages/block-editor/src/components/block-caption/index.native.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Caption, RichText } from '@wordpress/block-editor';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+
+const BlockCaption = ( { accessible, accessibilityLabel, onBlur, onChange, onFocus, isSelected, shouldDisplay, text } ) => (
+	<View
+		accessibilityLabel={ accessibilityLabel }
+		accessibilityRole="button"
+		accessible={ accessible }
+		style={ { padding: 12, flex: 1, display: shouldDisplay ? 'flex' : 'none' } }
+	>
+		<Caption
+			isSelected={ isSelected }
+			onBlur={ onBlur }
+			onChange={ onChange }
+			onFocus={ onFocus }
+			value={ text }
+		/>
+	</View>
+);
+
+export default compose( [
+	withSelect( ( select, { accessibilityLabelCreator, clientId } ) => {
+		const {
+			getBlockAttributes,
+			getSelectedBlockClientId,
+		} = select( 'core/block-editor' );
+		const { caption } = getBlockAttributes( clientId );
+		const accessibilityLabel = accessibilityLabelCreator ? accessibilityLabelCreator( caption ) : undefined;
```

Comment:
Not at all a blocker, just something I've been thinking about: I wonder if it makes sense to move this to the caption :thinking: . Also, I wonder if it needs to be set in `withSelect`.. perhaps it can be a regular prop. I understand we get `caption` from block attributes there, but I guess we could also use the `text` that we pass to caption. Wdyt?

Reply:
Great suggestion! I think we should do it, I'll move some things to the `Caption` component and test how it works with the accessibility changes 😃